modules/setupadBidAdapter.js

```diff
@@ -0,0 +1,420 @@
+      for (const bidderName in biddersCreativeIds) {
+        if (
+          biddersCreativeIds.hasOwnProperty(bidderName) &&
+          biddersCreativeIds[bidderName] === bid.creativeId
+        ) {
+          bidder = bidderName;
+          break; // Exit the loop if a match is found
+        }
+      }
+
+      // Add extra parameters
+      extraBidParams = `&cpm=${bid.originalCpm}&currency=${bid.originalCurrency}`;
+    }
+
+    const url = `${REPORT_ENDPOINT}?event=${eventName}&bidder=${bidder}&placementIds=${placementIds}&auctionId=${auctionId}${extraBidParams}&timestamp=${timestamp}`;
+
+    return url;
+  },
+};
+
+function getBidders(serverResponse) {
+  const bidders = serverResponse
+    .map((res) => Object.keys(res.body.ext.responsetimemillis || []))
+    .flat(1);
+
+  if (bidders.length) {
+    return encodeURIComponent(JSON.stringify([...new Set(bidders)]));
+  }
+}
+
+function getAdEl(bid) {
+  // best way I could think of to get El, is by matching adUnitCode to google slots...
+  const slot =
+    window.googletag &&
+    window.googletag.pubads &&
+    window.googletag
+      .pubads()
+      .getSlots()
+      .find((slot) => slot.getAdUnitPath() === bid.adUnitCode);
+  const slotElementId = slot && slot.getSlotElementId();
+  if (!slotElementId) return null;
+  return document.querySelector('#' + slotElementId);
+}
+
+function getBoundingClient(bid) {
+  const el = getAdEl(bid);
+  if (!el) return {};
+  return el.getBoundingClientRect();
+}
+
+function getAd(bid) {
+  let ad, adUrl, vastXml, vastUrl;
+
+  switch (deepAccess(bid, 'ext.prebid.type')) {
+    default:
+      if (bid.adm && bid.nurl) {
+        ad = bid.adm;
+        ad += createTrackPixelHtml(decodeURIComponent(bid.nurl));
+      } else if (bid.adm) {
+        ad = bid.adm;
+      } else if (bid.nurl) {
+        adUrl = bid.nurl;
+      }
+  }
+
+  return { ad, adUrl, vastXml, vastUrl };
+}
+
+function getSiteObj() {
+  const refInfo = (getRefererInfo && getRefererInfo()) || {};
+
+  return {
+    page: refInfo.page,
+    ref: refInfo.ref,
+    domain: refInfo.domain,
+  };
+}
+
+function getDeviceObj() {
+  return {
+    w:
+      window.innerWidth ||
+      window.document.documentElement.clientWidth ||
+      window.document.body.clientWidth ||
+      0,
+    h:
+      window.innerHeight ||
+      window.document.documentElement.clientHeight ||
+      window.document.body.clientHeight ||
+      0,
+  };
+}
+
+function initSendingDataStatistic() {
+  class SendingDataStatistic {
+    eventNames = [
+      CONSTANTS.EVENTS.BID_TIMEOUT,
+      CONSTANTS.EVENTS.BID_RESPONSE,
+      CONSTANTS.EVENTS.BID_REQUESTED,
+      CONSTANTS.EVENTS.NO_BID,
+      CONSTANTS.EVENTS.BID_WON,
+    ];
+
+    disabledSending = false;
+    enabledSending = false;
+    auctionIds = {};
+    eventHendlers = {};
+
+    initEvents() {
+      this.disabledSending =
+        !!config.getBidderConfig()?.setupad?.disabledSendingStatisticData;
```

Comment:
make sure you document this

Reply:
sure, in our docs we explain how to disable sending statistics data